### PR TITLE
[M5] Remove reliance on the OpenShiftClusterDocument in install code

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Part of https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/14625838

### What this PR does / why we need it:

Removes the need to know about OpenShiftClusterDocument in pkg/installer. Makes pulling it out easier.

### Test plan for issue:

E2E

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: pkg/installer/manager.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ type manager struct {
 	log *logrus.Entry
 	env env.Interface
 
-	doc             *api.OpenShiftClusterDocument
+	uuid            string
```
Comment (s-amann) [inline pkg/installer/manager.go:23]: Is there a more descriptive name for `uuid`? perhaps docId, or clusterDocId? I know we want to remove reliance on the doc, but what is this `uuid` for then?

**Review by s-amann — APPROVED** [review]
lgtm

**Review by s-amann — DISMISSED** [review]

**Review by petrkotas — APPROVED** [review]
Hi @hawkowl to clarify I understand. You have replaced internal OpenShiftClusterDocument with external API OpenshiftCluster.

Anyway thanks I believe it is good to go.

**Review by petrkotas — CHANGES_REQUESTED** [review]
Damn I forgot about my uuid comment. Would you please address it? Thank you.

**Review by darthhexx — APPROVED** [review]
Thanks for changing the uuid var name, LGTM.

**Pull request merged**